packages/quantic/force-app/main/default/lwc/quanticNumericFacet/__tests__/quanticNumericFacet.test.js

```diff
@@ -229,11 +407,442 @@ describe('c-quantic-numeric-facet', () => {
+      );
+      expect(placeholder).not.toBeNull();
+    });
+
+    describe('when the interface has an error', () => {
+      beforeEach(() => {
+        searchStatusState = {
+          ...initialFacetState,
+          isLoading: true,
+          hasError: true,
+          firstSearchExecuted: false,
+        };
+      });
+
+      it('should not display the placeholder component', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const placeholder = element.shadowRoot.querySelector(
+          selectors.placeholder
+        );
+        expect(placeholder).toBeNull();
+      });
+    });
+  });
+
+  describe('when the component is ready', () => {
+    describe('when the facet has values', () => {
+      beforeEach(() => {
+        facetState = {
+          ...initialFacetState,
+          values: exampleFacetValues,
+        };
+      });
+
+      it('should display the facet card when the facet values have results', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const cardContainer = element.shadowRoot.querySelector(
+          selectors.cardContainer
+        );
+        expect(cardContainer).not.toBeNull();
+      });
+
+      it('should not display the facet card when the facet values have no results', async () => {
+        facetState = {
+          ...initialFacetState,
+          values: [{start: 1, end: 2, numberOfResults: 0}],
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const cardContainer = element.shadowRoot.querySelector(
+          selectors.cardContainer
+        );
+        expect(cardContainer).toBeNull();
+      });
+
+      describe.each([
+        ['checkbox', 'toggleSelect'],
+        ['link', 'toggleSingleSelect'],
+      ])(
+        'when the property displayValuesAs is set to "%s"',
+        (propertyValue, expectedFunctionToBeCalled) => {
+          test(`should display the facet value as ${propertyValue}`, async () => {
+            const expectedFacetValues = exampleFacetValues.map(
+              (facetValue) => ({
+                ...facetValue,
+                checked: false,
+                key: `${facetValue.start}..${facetValue.end}`,
+              })
+            );
+            const element = createTestComponent({
+              ...defaultOptions,
+              displayValuesAs: propertyValue,
+            });
+            await flushPromises();
+
+            const facetValueElements = element.shadowRoot.querySelectorAll(
+              selectors.facetValue
+            );
+            expect(facetValueElements.length).toBe(exampleFacetValues.length);
+            expectedFacetValues.forEach((facetValue, index) => {
+              expect(facetValueElements[index].item).toEqual(facetValue);
+              expect(facetValueElements[index].isRangeFacet).toEqual(true);
+              expect(facetValueElements[index].displayAsLink).toEqual(
```

Comment:
clever ;)

nitpick, but since this is within a forEach loop and also because it might make it more readable, I think I would do a 

`const shouldDisplayValueAsLink = propertyValue === 'link'` outside of the `forEach`, that way it's evaluated only once instead of for each facet values.

BTW you could also add that within your `.each`, and you can even name them, and you can use the functions directly instead of only their name I think.

```
[displayValuesAs: 'checkbox', expectedFunctionToBeCalled: functionsMock.toggleSelect, shouldDisplayAsLink: false],
[displayValuesAs: 'link', expectedFunctionToBeCalled: functionsMock.toggleSingleSelect, shouldDisplayAsLink: true],
```